### PR TITLE
[chore] Move newNodeSharedInformer to a common/shared pkg

### PR DESCRIPTION
**Pull request by ChrsMark**
**Description:** <Describe what has changed.>

This PR implements some basic refactoring required for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32295.
Doing this in a standalone PR was requested at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32295#issuecomment-2064312509.

The change moves a library method called `newNodeSharedInformer` from the `k8sattributes` processor package to a generic one since it can be re-used by multiple various components. 

Note: I'm not sure if a changelog is required for this change, so I left it out for now.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by ChrsMark** [comment]
I'm not sure if the failure is related or it is some "flakiness" 🤔 .
Locally the test passes:

```console
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer	15.464s
```

**Review by TylerHelmuth — APPROVED** [review]

**Comment by ChrsMark** [comment]
@open-telemetry/collector-contrib-approvers any change I could get more eyes on this please?
This one is required to unblock https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32295.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
